src/plugin/src/main/java/com/defenseunicorns/uds/keycloak/plugin/authentication/RequireGroupAuthenticator.java

```diff
@@ -121,11 +119,49 @@ private String buildGroupPath(GroupModel group) {
         return path.toString();
     }
 
-    private void success(final AuthenticationFlowContext context, final UserModel user) {
-        user.addRequiredAction("TERMS_AND_CONDITIONS");
+    protected void success(final AuthenticationFlowContext context, final UserModel user) {
+        boolean shouldAddTAC = true;
+
+        if (isConditionalTACActive(context)) {
+            String parentSessionId = context.getAuthenticationSession().getParentSession().getId();
+            LOGGER.debugf("Conditional TAC, parent session ID: %s", parentSessionId);
+            if (user.getAttributes().get(TAC_USER_ATTRIBUTE) != null) {
+                String userSessionId = user.getAttributes().get(TAC_USER_ATTRIBUTE).get(0);
+                LOGGER.debugf("Conditional TAC, user session ID: %s", parentSessionId);
+                if (parentSessionId.equals(userSessionId)) {
+                    shouldAddTAC = false;
+                    LOGGER.info("User already accepted Terms and Conditions for this session. Skipping...");
+                } else {
+                    LOGGER.info("Stale session detected, asking user to accept Terms and Conditions again");
+                }
+            } else {
+                LOGGER.info("User didn't accept Terms and Conditions. Requesting accepting it");
```

Comment:
```suggestion
                LOGGER.info("User hasn't accepted Terms and Conditions. Requesting user accept them.");
```

nit: i think the wording is a little confusing